airbyte-workers/src/main/java/io/airbyte/workers/temporal/ConnectionManagerUtils.java

```diff
@@ -136,30 +156,56 @@ static ConnectionManagerWorkflow startConnectionManagerNoSignal(final WorkflowCl
    * @param connectionId the ID of the connection whose workflow should be retrieved
    * @return the healthy ConnectionManagerWorkflow
    * @throws DeletedWorkflowException if the workflow was deleted, according to the workflow state
-   * @throws UnreachableWorkflowException if the workflow is unreachable
+   * @throws UnreachableWorkflowException if the workflow is in an unreachable state
    */
   static ConnectionManagerWorkflow getConnectionManagerWorkflow(final WorkflowClient client, final UUID connectionId)
       throws DeletedWorkflowException, UnreachableWorkflowException {
+
     final ConnectionManagerWorkflow connectionManagerWorkflow;
     final WorkflowState workflowState;
+    final WorkflowExecutionStatus workflowExecutionStatus;
     try {
       connectionManagerWorkflow = client.newWorkflowStub(ConnectionManagerWorkflow.class, getConnectionManagerName(connectionId));
       workflowState = connectionManagerWorkflow.getState();
+      workflowExecutionStatus = getConnectionManagerWorkflowStatus(client, connectionId);
     } catch (final Exception e) {
       throw new UnreachableWorkflowException(
           String.format("Failed to retrieve ConnectionManagerWorkflow for connection %s due to the following error:", connectionId),
           e);
     }
 
-    if (workflowState.isDeleted()) {
-      throw new DeletedWorkflowException(String.format(
-          "The connection manager workflow for connection %s is deleted, so no further operations cannot be performed on it.",
-          connectionId));
+    if (WorkflowExecutionStatus.WORKFLOW_EXECUTION_STATUS_COMPLETED.equals(workflowExecutionStatus)) {
+      if (workflowState.isDeleted()) {
+        throw new DeletedWorkflowException(String.format(
+            "The connection manager workflow for connection %s is deleted, so no further operations cannot be performed on it.",
+            connectionId));
+      }
+
+      // A non-deleted workflow being in a COMPLETED state is unexpected, and should be corrected
+      throw new UnreachableWorkflowException(
+          String.format("ConnectionManagerWorkflow for connection %s is unreachable due to having COMPLETED status.", connectionId));
```

Comment:
Added this case because when testing I saw an occurrence of a workflow with COMPLETED status that was not deleted. Since it was COMPLETED, it's state could still be queried, but no signals could be executed on it. This logic corrects this by correctly treating workflows in this state as "unreachable"